modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/rel/IgniteTableModify.java

```diff
@@ -115,10 +114,9 @@ public IgniteRel clone(RelOptCluster cluster, List<IgniteRel> inputs) {
 
     @Override
     public RelWriter explainTerms(RelWriter pw) {
+        // for correct rel obtaining from ExecutionServiceImpl#physNodesCache.
```

Comment:
To be honest, I don't understand what the problem is. Could you provide more details in the comment?

Reply:
We can hold mapping on outdated rels, mute this fix and check: ItJoinTest#testIsNotDistinctFrom for example, or probably we need to invalidate this cache more carefully.